lib/tcpip/network-internal.h

```diff
@@ -46,7 +46,7 @@ __cheri_compartment("TCPIP") int network_host_resolve(
  *
  * This returns a sealed capability to a socket on success, or null on failure.
  *
- * This should be called only from the NetAPI compartment.
+ * This should be called only from the NetAPI or TCP/IP compartments.
```

Comment:
Why is it called from the TCP/IP compartment? Wouldn’t that use internal APIs?

This contradicts the property enforced in rego:
https://github.com/CHERIoT-Platform/network-stack/blob/473249a6ace45f1d1fe9a1606705355b190327b3/network_stack.rego#L167

Reply:
We use it in `network_socket_udp`, which is just a light wrapper for `network_socket_create_and_bind` with connection type UDP. I'm not sure using an internal API is worth it, the current code makes sense. We should update the rego as well, will update.